src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+        &trajectory, segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    app_trajectory_planner_generatePositionTrajectoryTimeProfile(&trajectory,
+                                                                 segment_lengths);
+
+    EXPECT_FLOAT_EQ(trajectory.trajectory_elements[0].time, 0.0);
+    EXPECT_NEAR(trajectory.trajectory_elements[1].time, 0.1818, 0.0001);
+    for (unsigned int i = 2; i < path_parameters.num_segments - 1; i++)
+    {
+        EXPECT_EQ(trajectory.trajectory_elements[i].time,
+                  trajectory.trajectory_elements[i - 1].time +
+                      segment_lengths[i - 1].angular_segment_length /
+                          path_parameters.max_allowable_angular_speed);
+    }
+    const float radians_moved_in_acceleration_phase =
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].orientation -
+        trajectory.trajectory_elements[path_parameters.num_segments - 2].orientation;
+
+    EXPECT_NEAR(trajectory.trajectory_elements[path_parameters.num_segments - 1].time,
+                9.1818, 0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_rebalance_trajectory_segment_linear)
+{
+    // Create a scenario where the linear path parameters will limit the speed of the
+    // trajectory by using angular parameters that are much greater
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.max_allowable_angular_acceleration = 1;
+    path_parameters.max_allowable_angular_speed        = 1;
+    path_parameters.max_allowable_linear_acceleration  = 1;
+    path_parameters.max_allowable_linear_speed         = 1;
+    path_parameters.num_segments                       = 100;
+    path_parameters.initial_linear_speed               = 1;
+    path_parameters.final_linear_speed                 = 1;
+    path_parameters.t_start                            = 1;
+    path_parameters.t_end                              = 10;
+    path_parameters.path                = {.x = {0, 0, 1, 0}, .y = {0, 0, 1, 0}};
+    path_parameters.orientation_profile = {.coefficients = {0, 0, 1, 0}};
+
+    float linear_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float angular_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    TrajectorySegment_t segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    PositionTrajectoryElement_t trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    PositionTrajectory_t trajectory;
+    trajectory.trajectory_elements = trajectory_elements;
+
+    trajectory.path_parameters = path_parameters;
+
+    // Create an artificial trajectory here for testing
+    trajectory.trajectory_elements[2].linear_speed = 1;
+    trajectory.trajectory_elements[3].linear_speed = 4;
+    float desired_time                             = 0.01;  // seconds
+
+    segment_lengths[2].linear_segment_length = 1.0;  // meters
```

Comment:
Can we move the end-of-line comments to the line above, or better yet make the units a part of the variable names? They're fine here, but I'd like to avoid setting the example of putting comments at the end of lines because they can get really knackered by clang-format, particularly on longer lines.

Reply:
Moved units to name